src/media/writer.rs

```diff
@@ -33,8 +33,14 @@ impl<'a> Writer<'a> {
     /// Get the configured payload parameters for the `mid` this writer is for.
     ///
     /// For the [`Writer::write()`] call, the `pt` must be set correctly.
-    pub fn payload_params(&self) -> &[PayloadParams] {
-        &self.session.codec_config
+    pub fn payload_params(&self) -> Vec<&PayloadParams> {
+        let media = self.session.media_by_mid(self.mid).unwrap();
+        let all_params = self.session.codec_config.params();
+        Vec::from_iter(
+            all_params
+                .iter()
+                .filter(|p| media.remote_pts().contains(&p.pt)),
+        )
```

Comment:
Ouch, this allocation is not so nice. Depends a bit on how the API is used.

I guess the only alternative is `-> impl Iterator<Item=&PayloadParams>`. Is that too hard for the user to parse though?

The other option is to duplicate the `PayloadParams` into `Media` to have the Vec ready already. Though I'm quite keen on not doing that now that finally got the definitions centrally in `Session`.

Thoughts @k0nserv ?

Reply:
I think your options are what we have to consider, i.e. an iterator or pre-computing the value. I can't think of a different solve to that